apps/extension/src/background/keyring/keyring.ts

```diff
@@ -450,16 +432,16 @@ export class KeyRing {
     }
 
     const timestamp = source === "generated" ? new Date().getTime() : 0;
-    const derivedAccount = await this.persistAccount(
-      type === AccountType.ShieldedKeys ? zip32Path : derivationPath,
+    const derivedShieldedAccount = await this.persistAccount(
+      path,
       parentId,
       type,
       alias,
       info,
       source,
       timestamp
     );
-    return derivedAccount;
+    return derivedShieldedAccount;
```

Comment:
Just a note about this change: So we had a `deriveAccount` method, which would alternatively call `deriveTransparentAccount` or `deriveShieldedAccount`, but it was never used for transparent, so I consolidated it into one function, `deriveShieldedAccount`

Reply:
If/when we allow deriving only Transparent, we can add a specific call for that at that time